pydm/data_plugins/local_plugin.py

```diff
@@ -1,332 +1,427 @@
+        None.
+
+        """
+        if enum_string is not None:
             try:
-                spec = inspect.getargspec(getattr(obj, self.attr))
-                self.nargs = len(spec.args) - 1
-            except:
-                self.nargs = None
-            if self.refresh > 0:
-                self.update_timer = QTimer()
-                self.update_timer.timeout.connect(self.update)
-                self.update_timer.start(1000.0 * self.refresh)
-            self.add_listener(channel)
-
-        def get_value(self):
-            """
-            Return the current value of this connection.
-
-            :rtyp: Can be any type, user defined.
-            """
-            attr = getattr(self.obj, self.attr)
+                self._enum_string = tuple(enum_string)
+                self.enum_strings_signal.emit(self._enum_string)
+            except ValueError:
+                logger.debug("Error when converting enum_string")
+
+    def format_type_params(self, type_kwargs):
+        """
+        Format value_type parameters.
+
+        Parameters
+        ----------
+        type_kwargs : dict
+
+        Returns
+        -------
+            dict
+
+        """
+
+        dtype = type_kwargs.get('dtype')
+
+        if dtype is not None:
+            dtype = dtype[0]
             try:
-                args = self.args
-                kwargs = self.kwargs
-                return attr(*args, **kwargs)
-            except AttributeError:
-                return attr
-
-        @Slot()
-        def update(self):
-            """
-            Get a new value from the object and send it to all listeners.
-            If an exception was thrown, send a disconnected signal.
-            """
+                self._type_kwargs['dtype'] = np.dtype(dtype)
+                return self._type_kwargs
+            except ValueError:
+                logger.debug('Cannot convert dtype')
+        else:
+            dtype = 'object'
             try:
-                value = self.get_value()
-            except:
-                self.send_connection_state(False)
-                return
-            self.send_connection_state(True)
-            self.send_new_value(value)
-
-        def send_new_value(self, value=None):
-            """
-            Send a value to every channel listening for our obj.
-
-            :param value: Value to emit to our listeners.
-            :type value:  int, float, str, or np.ndarray.
-            """
-            if isinstance(value, np.generic):
-                value = np.asscalar(value)
-            if isinstance(value, np.ndarray):
-                self.new_waveform_signal.emit(value)
-            elif isinstance(value, (int, float, str)):
-                self.new_value_signal[type(value)].emit(value)
-            else:
-                self.new_value_signal[str].emit(str(value))
+                self._type_kwargs['dtype'] = np.dtype(dtype)
+                return self._type_kwargs
+            except ValueError:
```

Comment:
```suggestion
            except TypeError:
```

`np.dtype` raises `TypeError` for bad input, it seems (try it yourself to confirm)